homeassistant/helpers/script.py

```diff
@@ -99,13 +101,55 @@ def script_delay(now):
 
                 self._async_unsub_delay_listener = \
                     async_track_point_in_utc_time(
-                        self.hass, script_delay,
+                        self.hass, async_script_delay,
                         date_util.utcnow() + delay)
                 self._cur = cur + 1
                 if self._change_listener:
                     self.hass.async_add_job(self._change_listener)
                 return
 
+            elif CONF_WAIT in action:
+                # Call ourselves in the future to continue work
+                wait = action[CONF_WAIT]
+                wait.hass = self.hass
+
+                @callback
+                def async_script_wait(entity_id, from_s, to_s):
```

Comment:
I think that we should not keep this inside the script but instead move to the `event` helper and call it `track_template`:

```python
async_track_template(hass, template, action)
```

Because this logic is very much alike the logic we have in `automation/template.py` and making it more generic would benefit more people.

(This can be done in another PR)